### ISSUE TITLE
Feature Request - A Case to Implement CraftTweaker Support. 

### ISSUE THREAD

**Issue by XvSkItZvX** [BloodWorkXGaming/ExNihiloCreatio#38]
I would like to begin that I am aware that "This means no more reliance on MineTweaker/ModTweaker." is a selling point for the mod, and in no way wish to make CraftTweaker a requirement for Ex Nihilo Creatio.

For individuals who like to simply throw a few mods in a folder, and make a few smaller tweaks, the native JSON support should suffice, and likely preferred. I tend to play larger packs and have hopes of making comparable packs of my own. It's not uncommon for modern packs to be near or exceed 200 mods.

With that in mind having to 'sift' (pun intended lol) through each mod that provides a plethora of various customization options becomes bothersome, even confusing, and increases the likelihood of mistakes. CraftTweaker has undergone quite a few improvements recently and provides a consolidated, versatile, and focused approach to recipe manipulation.

The more mods that provide compatibility with CraftTweaker reduces the work load on pack developers. Being able to control a number of aspects from different mods within a single script is an invaluable asset to pack developers.

When paired with other 'tweaker' styled mods, B.A.S.E / ContentTweaker specifically, new resource can be created then have, oreDict entry, means of acquisition, processing, and uses assigned to it in a single *.Zs file. And in the very near future, if I understand correctly, we will be able to provide a custom model/texture for the items. 

I cannot speak for everyone, and would like to make it clear that just a few moths ago I was looking for a way to make modifications to recipes for unique crafting methods provided by mods without using CraftTweaker. Because I was intimidated by it and didn't want to 'F* Something up'. I have since taken the opportunity to learn more about it, and play with it some, though still not proficient, and came to believe that any mod that provides a unique crafting method should also make use of the CraftTweaker framework. 

There are a few mods that have provided a means of abstracting crafting mechanics, "Extended Crafting" and "Modular Machinery", are the two that immediately come to mind. Of which, Extended Crafting has implemented the Crafting Tweaks Framework, and I intend to approach Modular Machines developer after he has had more time to refine his original concept, about making use of the framework. 

I thank you for your time in reading this and the great amount of effort your put into contributing excellent content to the community. I do want you to know that your work is greatly appreciated, and that my intent is not to critique a mod I feel is done wrong, but to help suggest an improvement on a mod that is already freaking awesome. Some of the potential I see in this mod is more than just a sky block approach but a viable path for a survival focused early game tier. And on that note I hope this will be well received have just noticed you recently forked ModTweaker, I may be doing little more than reaffirming that you've made a good decision lol...

**Comment by BloodWorkXGaming** [comment]
First of all thanks for taking the time writing the Suggestion, probably the longest I've ever gotten :P
And as I did a lot of PRs to CraftTweaker I indeed know what it could offer.
Most notebly the new preprocessor stuff which allows packmodes could be pretty powerful, would make it easier than having 20 different config folders for different packmodes.
Might be clearer to read as well.

The only issue I see is in maintaining multiple version of a config system, which adding CrT support would mean.

**Comment by BloodWorkXGaming** [comment]
Oh, another thing that i can do in json but not in the CrT is the reload command.

**Comment by XvSkItZvX** [comment]
Thank you for replying, I was a bit concerned about whether or not implementing an additional system would provide enough of a benefit to out weigh the additional work. I tend to use the External Tweaker mod/app thus am not partial about the reload command, but, I can see the benefit of making recipe modifications in a live environment and simply reload as opposed to having to restart the entire client. Would allow a user to adjust a single recipe on the fly. Thanks again for your time and I look forward to seeing what type of creative uses I can find for this mod.

**Comment by BloodWorkXGaming** [comment]
I got started on adding compat for the Sieve, it should not break too much, as it is just getting loaded together with the json recipes and therefore should work alongside it.

Slight problem here.... there are like 15 registries I would have to add support to and ye, kinda too lazy/don't really have the time to add them all by hand, so If you have any coding experience and you want to continue it for the others I would be happy to accept a PR ^^

**Comment by XvSkItZvX** [comment]
I would love to, even gone as far as watching any videos I can find on the topic, and reading books, and I can make very crude rudimentary programs in C++. I know data types, and the difference between a long and a float, heck I can even say that pointers make sense to me and that there is only a small margin of variation between each language cause in the end they all send the same commands in machine language based on Turing Primitives, all the sources I get want to explain variables, and structures, and methods/functions/classes/objects, but I haven't found a good source to actually link it all together in a way that just "clicks" with me, and makes me understand the practicality and actually how to get from a couple of lines to an actual program. So sadly, even though I'm not completely lost understanding the concept of programming, I still have no talent for it. Honestly familiarizing myself with Git as I have and getting to understand the mod configurations is just another step before I start looking into other peoples sources and trying to figure it out from the inside. CoFH world gave me a reason to learn JSON, (more so than javaScript (ECMAScript lol) actually was. sadly though I haven't found out how to set up Eclipse, though maybe I should ask you about IntelliJ (assumption based on some of your activities on git) lol..

In short, I don't have the skill set needed, even though I do have all this time lol. Maybe in the future though..

**Comment by BloodWorkXGaming** [comment]
Ye no worries ^^
Can do it myself, will just take a bit longer then.
Oh and also, I didn't know my git history tells so much about me :P

**Comment by XvSkItZvX** [comment]
I tend to pay a lof of attention. It's part of your repositories under your profile lol... And speaking of attention, forgive me for being so forward and asking but I'm curious are you from Russia? You replied from your email instead of the site this time. Shows the details of the email am curious about "Okt" which I assume is the abbreviated month, and it's already tomorrow there lol.. In fairness I'm from the US.

**Comment by BloodWorkXGaming** [comment]
Na, I'm from Germany, but AFAIK I even have that somewhere on my github
Profile.
Okt -> Oktober

Now I am curious as well, do you see the github groups I am in or is that
private information? There would be a lot more information hidden back
there...

**Comment by XvSkItZvX** [comment]
You're right it does, I missed it, and had I been thinking I would have caught that Oktober is the German spelling, we see it over here a lot this time of the year with "Oktoberfest" celebrations. And didn't consider that IDL is GMT+12 so it would be noon, or afternoon, where I thought you were at the time. Below I took a screen cap of all pages related to your profile that I have access to. 

![profile](https://user-images.githubusercontent.com/9143369/31322130-4c7de1f8-ac57-11e7-9db6-149b613dcbca.PNG)
![repositories](https://user-images.githubusercontent.com/9143369/31322131-4c7ddbfe-ac57-11e7-9dce-1d1a9c3c559b.PNG)
![stars](https://user-images.githubusercontent.com/9143369/31322128-4c7da698-ac57-11e7-969e-08dace817cc7.PNG)
![followers](https://user-images.githubusercontent.com/9143369/31322129-4c7de162-ac57-11e7-97c7-0d88b50cc7c2.PNG)
![following](https://user-images.githubusercontent.com/9143369/31322127-4c7cd7c2-ac57-11e7-901e-ae8e9726a876.PNG)

Nothing too telling from the site itself and only slightly sensitive when you reply back through email. 

anyways I know it's late, thanks for your time I won't keep you up any longer lol.

**Comment by XvSkItZvX** [comment]
Quick question now that I'm working on the changes I am hoping to make to for the pack. Do the custom JSONs follow the same style as the ones used for Adscensio?

**Comment by BloodWorkXGaming** [comment]
Yup, I rewrote the code for them, however I left the internal syntax of the
registries the same

XvSkItZvX <notifications@github.com> schrieb am Sa., 14. Okt. 2017, 06:31:
